### PR TITLE
Porting srdfdom to ROS2

### PR DESCRIPTION
**Pull request by anasarrak**
Can we create a ros2 branch?

### CODE REVIEW

**Comment by davetcoleman** [comment]
Done: https://github.com/ros-planning/srdfdom/tree/ros2

**Comment by anasarrak** [comment]
@davetcoleman Are the changes ok for you? Can we merge it?

**Comment by davetcoleman** [comment]
There is a merge conflict with https://github.com/ros-planning/srdfdom/pull/44/files

File: include/srdfdom/model.h [hunk 1]
```diff
@@ -41,7 +41,7 @@
 #include <string>
 #include <vector>
 #include <utility>
-#include <urdf/model.h>  // TODO: replace with urdf_model/types.h in Lunar
+#include <urdf/model.h>
```
Comment (davetcoleman) [inline include/srdfdom/model.h:44]:
```suggestion
#include <urdf_model/types.h>
```

File: package.xml [hunk 2]
```diff
@@ -1,32 +1,39 @@
-<package>
+<?xml version="1.0"?>
+<?xml-model href="http://download.ros.org/schema/package_format3.xsd" schematypens="http://www.w3.org/2001/XMLSchema"?>
+<package format="3">
   <name>srdfdom</name>
   <version>0.5.1</version>
   <description>Parser for Semantic Robot Description Format (SRDF).</description>
   <author email="isucan@willowgarage.com">Ioan Sucan</author>
   <author email="gwalck@techfak.uni-bielefeld.de">Guillaume Walck</author>
+  <author email="vmayoral@acutronicrobotics.com">Víctor Mayoral Vilches</author>
```
Comment (davetcoleman) [inline package.xml]: I don't think porting to ROS2 warrants authorship, since no functionality has changed.

File: package.xml [hunk 3]
```diff
@@ -1,32 +1,39 @@
...
   <url type="website">http://ros.org/wiki/srdfdom</url>
   <url type="bugtracker">https://github.com/ros-planning/srdfdom/issues</url>
   <url type="repository">https://github.com/ros-planning/srdfdom</url>
-  
-  <buildtool_depend>catkin</buildtool_depend>
 
+  <buildtool_depend>ament_cmake</buildtool_depend>
   <build_depend>boost</build_depend>
-  <build_depend>cmake_modules</build_depend>
+  <!-- <build_depend>cmake_modules</build_depend> -->
```
Comment (davetcoleman) [inline package.xml]: remove

File: package.xml [hunk 4]
```diff
@@ -1,32 +1,39 @@
...
   <build_depend>libconsole-bridge-dev</build_depend>
-  <build_depend>urdf</build_depend>
-  <build_depend>liburdfdom-headers-dev</build_depend>
+  <build_depend>urdfdom_headers</build_depend>
+  <!-- <build_depend>liburdfdom-headers-dev</build_depend> -->
```
Comment (davetcoleman) [inline package.xml]: remove

File: package.xml [hunk 5]
```diff
@@ -1,32 +1,39 @@
...
-  <run_depend>libconsole-bridge-dev</run_depend>
-  <run_depend>liburdfdom-headers-dev</run_depend>
-  <run_depend>tinyxml</run_depend>
-  <run_depend>urdfdom_py</run_depend>
-
-  <test_depend>rostest</test_depend>
+  <exec_depend>urdf</exec_depend>
+  <exec_depend>boost</exec_depend>
+  <exec_depend>libconsole-bridge-dev</exec_depend>
+  <!-- <exec_depend>liburdfdom-headers-dev</exec_depend> -->
```
Comment (davetcoleman) [inline package.xml]: remove

File: src/srdf_writer.cpp [hunk 6]
```diff
@@ -150,10 +150,12 @@ TiXmlDocument SRDFWriter::generateSRDF()
...
-  TiXmlComment* comment =
-      new TiXmlComment("This does not replace URDF, and is not an extension of URDF.\n    This is a format for "
-                       "representing semantic information about the robot structure.\n    A URDF file must exist for "
-                       "this robot as well, where the joints and the links that are referenced are defined\n");
+  TiXmlComment* comment = new TiXmlComment("This does not replace URDF, and is not an extension of URDF.\n    This is "
+                                           "a format for "
+                                           "representing semantic information about the robot structure.\n    A URDF "
+                                           "file must exist for "
+                                           "this robot as well, where the joints and the links that are referenced are "
+                                           "defined\n");
```
Comment (davetcoleman) [inline src/srdf_writer.cpp]: this is a messy change that is not necessary, please revert.

**Comment by anasarrak** [comment]
@davetcoleman Addressed every suggestion, can you check it?

**Comment by anasarrak** [comment]
Reverted back the change https://github.com/ros-planning/srdfdom/pull/49/commits/9243d92682b751f16b4657629da934b9de7272dd, it breaks moveit_core/robot_model, it's not encountering **ModelInterface**

**Comment by rhaschke** [comment]
@henningkayser, I guess this PR can be closed given that we have merged #52?

**Pull request closed without merging**